psamm/command.py

```diff
@@ -91,6 +92,16 @@ def init_parser(cls, parser):
     def run(self):
         """Execute command"""
 
+    def argument_error(self, msg):
+        """Raise error indicating error parsing an argument."""
+        raise CommandError(msg)
+
+    def fail(self, exc=None):
```

Comment:
The `exc` parameter would be in instance of `Exception` indicating the exception that caused the command to fail. You can expand the docstring with an explanation of this.